app/components/govuk_component/task_list_component/status_component.rb

```diff
@@ -0,0 +1,25 @@
+module GovukComponent
+  class TaskListComponent::StatusComponent < GovukComponent::Base
+    attr_reader :text, :href, :colour
+
+    def initialize(text: nil, href: nil, colour: nil, classes: [], html_attributes: {})
+      @text = text
+      @href = href
+      @colour = colour
+
+      super(classes: classes, html_attributes: html_attributes)
+    end
+
+    def call
+      tag.div(**html_attributes) do
+        render(GovukComponent::TagComponent.new(text: text, colour: colour))
+      end
```

Comment:
Hmm, so one tricky thing with the new task list component is that using the tag component within the status is 'optional', and there also needs to be a way to just pass in plain text or html. This is recommended for 'Completed' items, where there’s no need for a coloured tag, and instead just the plain text of "Completed" can be used.

In the Nunjucks macro, this is done by having an optional `tag` object within the `status` object, like this:

```javascript
items: [
  {
    title:
      {
        text: "Company Directors"
      },
    href: "#",
    status: {
      text: "Completed"
    }
  },
  {
    title:
      {
        text: "Registered company details"
      },
    href: "#",
    status: {
      tag: {
        text: "Incomplete",
        classes: "govuk-tag--blue"
      }
    }
  }
]
```

for the Ruby component, maybe we could do something similar, like this:

```ruby
<%= govuk_task_list do |task_list|
  task_list.with_item(title: "Contact details", href: '#', status: "Completed")
  task_list.with_item(title: "Project details", href: '#', status: {tag: {text: "Incomplete", colour: "green"}})
end %>
```

Or alternatively we don't do anything smart with the arguments, and instead get people to use the tag helper directly:

```ruby
<%= govuk_task_list do |task_list|
  task_list.with_item(title: "Contact details", href: '#', status: "Completed")
  task_list.with_item(title: "Project details", href: '#', status: govuk_tag(text: "Incomplete", colour: "green"))
end %>
```

...which might be simpler?

Reply:
Ah, I didn't realise regular text was acceptable too - I like the second option.

Reply:
I had a play with this and passing in raw HTML like `status: govuk_tag(...)` and we need to ensure it's safe, which here adds some risk.

I think this needs to be done via the block:

```erb
<%= govuk_task_list do |task_list|
  task_list.with_item(title: "Project details", href: '#') do |item|
    item.with_title(text: "...", href: "...", hint: "...")
    item.with_status { govuk_tag(text: "Incomplete", colour: "green")) }
  end
end %>
```

Reply:
You’ve got 2 titles and hrefs there…

I forget exactly how the html_safe system works in Rails, but I do think the  params version is easier to use than the block one, if we can figure it out.

Reply:
Oops, I forgot to remove the args, I've updated the snippet.

The problem here is that we need to call `#html_safe` on the `status` param, but that could contain anything, like if a team made dynamic statuses for example.